kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/module/GodotModuleBuilder.kt

```diff
@@ -0,0 +1,184 @@
+package godot.intellij.plugin.module
+
+import com.intellij.ide.util.projectWizard.ModuleBuilder
+import com.intellij.ide.util.projectWizard.ModuleBuilderListener
+import com.intellij.ide.util.projectWizard.ModuleWizardStep
+import com.intellij.ide.util.projectWizard.WizardContext
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.fileChooser.FileChooserDescriptor
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.module.ModuleType
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.openapi.util.IconLoader
+import com.intellij.openapi.util.Key
+import com.intellij.ui.layout.GrowPolicy
+import com.intellij.ui.layout.panel
+import com.intellij.ui.layout.selected
+import godot.intellij.plugin.GodotPluginBundle
+import java.io.File
+import javax.swing.JCheckBox
+import javax.swing.JComponent
+import javax.swing.JTextField
+
+class GodotModuleBuilder : ModuleBuilder(), ModuleBuilderListener {
+    private val groupIdKey = Key<String>("GROUP_ID")
+    private val artifactIdKey = Key<String>("ARTIFACT_ID")
+    private val versionKey = Key<String>("VERSION")
+    private val kotlinVersionKey = Key<String>("KOTLIN_VERSION")
+    private val godotKotlinJvmVersionKey = Key<String>("GODOT_KOTLIN_JVM_VERSION")
+    private val androidEnabledKey = Key<Boolean>("ANDROID_ENABLED")
+    private val dxToolPathKey = Key<String>("DX_TOOL_PATH")
+
+    private lateinit var wizardContext: WizardContext
+
+    private val builderName = GodotPluginBundle.message("module.builderName")
+    private val groupName = GodotPluginBundle.message("module.groupName")
+
+    init {
+        addListener(this)
+    }
+
+    override fun getBuilderId() = "godot.module.builder"
+    override fun getName() = builderName
+
+    override fun getModuleType(): ModuleType<*> {
+        return GodotModuleType
+    }
+
+    override fun getPresentableName() = GodotPluginBundle.message("module.presentableName")
+    override fun getDescription() = GodotPluginBundle.message("module.description")
+    override fun getNodeIcon() = IconLoader.getIcon("/pluginIcon_small.svg", this::class.java)
+    override fun getGroupName() = groupName
+
+    override fun getCustomOptionsStep(context: WizardContext, parentDisposable: Disposable): ModuleWizardStep {
+        wizardContext = context
+        return object : ModuleWizardStep() {
+            val groupIdTextField = JTextField("com.example")
+            val artifactIdTextField = JTextField("game")
+            val versionTextField = JTextField("0.0.1-SNAPSHOT")
+            val kotlinVersionTextField = JTextField("1.4.31")
+            val godotKotlinJvmVersionTextField = JTextField("0.1.0-3.2")
```

Comment:
I think we should rely on official tags like 3.2.3.

Reply:
Indeed. I'll update all of the versions in #109 as well

Reply:
The plugin is built at the same time as the rest of the bindings, so I think the choice of the bindings version shouldn't exist and be hardcoded in the plugin.
Nothing guarantee that the project template will stay stable in the future (because of this project and because of Godot as well).
Let's say we later update to Godot 4.0, if the user doesn't update the plugin to the new version, there is a good chance nothing will work if they target the 4.0 version of our bindings with a old plugin.

Reply:
Still everything will work. Just the plugin may act crazy. But that's the case for all Intellij plugin's that add framework support (for example spring boot, or sqldelight).
You always have to make sure you're using the correct version of the plugin if you use it together with a specific framework version.

Reply:
But yeah i can of course remove this option alltogether